internal/oidc/keyhandler.go

```diff
@@ -133,26 +133,39 @@ func (h *keyHandler) getKeySet() jwk.Set {
 	return h.keySet
 }
 
-func (h *keyHandler) getKeyFromID(ctx context.Context, keyID string) (jwk.Key, error) {
+func (h *keyHandler) getKeyFromID(ctx context.Context, keyID string, algorithm jwa.SignatureAlgorithm) (jwk.Key, error) {
 	keySet := h.getKeySet()
 
+	key, err := findKey(ctx, keyID, algorithm, keySet)
+	if err == nil {
+		return key, err
```

Comment:
error is nil, but returning error (linter will warn).

```suggestion
		return key, nil
```